From 13142f38241f1025053109d6ccbaab591cd82d85 Mon Sep 17 00:00:00 2001
From: Igor Opaniuk <igor.opaniuk@foundries.io>
Date: Tue, 20 Jul 2021 17:55:02 +0300
Subject: [PATCH 160/322] [FIO internal] imx8qm_mek: adjust loadaddr and
 scriptaddr

Adjust scriptaddr for all IMX8QM_MEK parts to 0x83100000.

Set loadaddr to custom address:
For TARGET_IMX8QM_MEK_A72_ONLY - 0xC7000000
For other TARGET_IMX8QM_MEK*   - 0x87000000

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
---
 include/configs/imx8qm_mek.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/configs/imx8qm_mek.h b/include/configs/imx8qm_mek.h
index 867d445ace4..06532993a9f 100644
--- a/include/configs/imx8qm_mek.h
+++ b/include/configs/imx8qm_mek.h
@@ -175,6 +175,7 @@
 	XEN_BOOT_ENV \
 	JAILHOUSE_ENV\
 	AHAB_ENV \
+	"scriptaddr=0x83100000\0" \
 	"script=boot.scr\0" \
 	"image=Image\0" \
 	SPLASH_IMAGE_ADDR \
@@ -259,11 +260,11 @@
 
 /* Link Definitions */
 #ifdef CONFIG_TARGET_IMX8QM_MEK_A72_ONLY
-	#define CONFIG_LOADADDR			0xC0280000
+	#define CONFIG_LOADADDR			0xC7000000
 	#define CONFIG_SYS_LOAD_ADDR		CONFIG_LOADADDR
 	#define CONFIG_SYS_INIT_SP_ADDR		0xC0200000
 #else
-	#define CONFIG_LOADADDR			0x80280000
+	#define CONFIG_LOADADDR			0x87000000
 	#define CONFIG_SYS_LOAD_ADDR		CONFIG_LOADADDR
 	#define CONFIG_SYS_INIT_SP_ADDR		0x80200000
 #endif
-- 
2.34.1

